File: proto_bench/bench.go
package main

import (
	"time"
	"io/ioutil"
	"github.com/golang/protobuf/proto"
	msg "./tutorial"
	"fmt"
)

func main() {
	content, _ := ioutil.ReadFile("./bin")
	message := &msg.Msg {
		Name: "Hello",
		Id: 0,
		Content: content,
	}
	start := time.Now()
	data, _ := proto.Marshal(message)

	newMessage := &msg.Msg{}

	proto.Unmarshal(data, newMessage)
	elapsed := time.Now().Sub(start)
	fmt.Println(elapsed)
}

File: proto_bench/bench.py
import time
import msg_pb2

file = open("./bin", "rb")

content = file.read()

file.close()

message = msg_pb2.msg()
message.name = "Hello"
message.id = 0
message.content = content

start = time.time()

data = message.SerializeToString()
new_message = msg_pb2.msg()
new_message.ParseFromString(data)

end = time.time()

print(end - start, "s.")
